IdentityCore/src/cache/crypto/MSIDAssymetricKeyLookupAttributes.m

```diff
@@ -25,46 +25,43 @@
-    keyPairAttr[(__bridge id)kSecAttrIsPermanent] = @YES;
-    keyPairAttr[(__bridge id)kSecAttrKeyType] = (__bridge id)kSecAttrKeyTypeRSA;
-    keyPairAttr[(__bridge id)kSecAttrKeySizeInBits] = @2048;
-    keyPairAttr[(__bridge id)kSecAttrLabel] = self.keyDisplayableLabel;
-    keyPairAttr[(__bridge id)kSecAttrAccessible] = (__bridge id)kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly;
+    NSData *tag = [self.privateKeyIdentifier dataUsingEncoding:NSUTF8StringEncoding];
+    NSDictionary* attributes =
+        @{ (id)kSecAttrKeyType:               (id)kSecAttrKeyTypeRSA,
+           (id)kSecAttrKeySizeInBits:         @2048,
+           (id)kSecAttrLabel:self.keyDisplayableLabel,
+           (id)kSecPrivateKeyAttrs:
+               @{ (id)kSecAttrIsPermanent:    @YES,
+                  (id)kSecAttrApplicationTag: tag,
+                  (id)kSecAttrAccessible:(id)kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly,
+                  (id)kSecAttrIsExtractable:@NO,
+                  (id)kSecAttrIsSensitive:@YES
+                  },
+         };
     
-    NSMutableDictionary *privateKeyAttr = [NSMutableDictionary new];
-    privateKeyAttr[(__bridge id)kSecAttrApplicationTag] = [self.privateKeyIdentifier dataUsingEncoding:NSUTF8StringEncoding];
-    privateKeyAttr[(__bridge id)kSecAttrIsSensitive] = @YES;
-    privateKeyAttr[(__bridge id)kSecAttrIsExtractable] = @NO;
-
-    NSMutableDictionary *publicKeyAttr = [NSMutableDictionary new];
-    publicKeyAttr[(__bridge id)kSecAttrApplicationTag] = [self.publicKeyIdentifier dataUsingEncoding:NSUTF8StringEncoding];
-    publicKeyAttr[(__bridge id)kSecAttrIsExtractable] = @YES;
-    
-    keyPairAttr[(__bridge id)kSecPrivateKeyAttrs] = privateKeyAttr;
-    keyPairAttr[(__bridge id)kSecPublicKeyAttrs] = publicKeyAttr;
-    
-    return keyPairAttr;
+    return attributes;
 }
 
+/*
+This particular query only queries the private key from the keychain and uses
+SecKeyCopyPublicKey(privateKey) to query the corresponding the public key.
+*/
 - (NSDictionary *)privateKeyAttributes
 {
-    NSMutableDictionary *keyAttr = [NSMutableDictionary new];
-    keyAttr[(__bridge id)kSecAttrApplicationTag] = [self.privateKeyIdentifier dataUsingEncoding:NSUTF8StringEncoding];
-    keyAttr[(__bridge id)kSecClass] = (__bridge id)kSecClassKey;
-    keyAttr[(__bridge id)kSecAttrKeyType] = (__bridge id)kSecAttrKeyTypeRSA;
-    return keyAttr;
-}
-
-- (NSDictionary *)publicKeyAttributes
-{
-    NSMutableDictionary *keyAttr = [NSMutableDictionary new];
-    keyAttr[(__bridge id)kSecAttrApplicationTag] = [self.publicKeyIdentifier dataUsingEncoding:NSUTF8StringEncoding];
-    keyAttr[(__bridge id)kSecClass] = (__bridge id)kSecClassKey;
-    keyAttr[(__bridge id)kSecAttrKeyType] = (__bridge id)kSecAttrKeyTypeRSA;
-    return keyAttr;
+    NSDictionary *getQuery = @{ (id)kSecClass: (id)kSecClassKey,
+                                (id)kSecAttrApplicationTag: [self.privateKeyIdentifier dataUsingEncoding:NSUTF8StringEncoding],
```

Comment:
Same here, this will crash if self.privateKeyIdentifier is nil.

Reply:
Addressed.